docs/SelectField.md

```diff
@@ -0,0 +1,44 @@
+# Select Field API
+
+This is a rich select field, allowing to show errors,
+hint texts and well as validity states like valid, warning and error.
+The `list` property supports a second level, which will be rendered as
+optgroup. Right now, the parent option will be rendered as a choosalbe option
+as well and it's label will be the label for the optgroup.
+
+## API
+
+This is a list of supported prop types:
```

Comment:
Great! But documenting APIs manually becomes stale quickly.

We can generate it to show it with Storybook as a tab: https://github.com/hipstersmoothie/storybook-addon-react-docgen (would be in a different PR)

Reply:
Oh, very nice tool! More value with less effort :heart: good fine!